build failed on osx with clang

tag: v1.1.54310-testing, tag: v1.1.54310-stable

Build commands:

```
mkdir build
cd build
cmake ..
make -j 4 clickhouse
```

The building process is good without errors, but the linking process fail on last step:

```
[100%] Building CXX object dbms/src/Server/CMakeFiles/clickhouse-server.dir/StatusFile.cpp.o
[100%] Building CXX object dbms/src/Server/CMakeFiles/clickhouse-server.dir/TCPHandler.cpp.o
[100%] Linking CXX static library libclickhouse-server.a
[100%] Built target clickhouse-server
[100%] Building CXX object dbms/src/Server/CMakeFiles/clickhouse-local.dir/LocalServer.cpp.o
[100%] Linking CXX static library libclickhouse-client.a
[100%] Built target clickhouse-client
[100%] Linking CXX static library libclickhouse-local.a
[100%] Built target clickhouse-local
[100%] Building CXX object dbms/src/Server/CMakeFiles/clickhouse.dir/main.cpp.o
[100%] Linking CXX executable clickhouse
Undefined symbols for architecture x86_64:
  "_current_memory_tracker", referenced from:
      DB::AsynchronousBlockInputStream::readImpl() in libclickhouse-client.a(Client.cpp.o)
      DB::AsynchronousBlockInputStream::next() in libclickhouse-client.a(Client.cpp.o)
      DB::AsynchronousBlockInputStream::calculate(MemoryTracker*) in libclickhouse-client.a(Client.cpp.o)
      DB::ProcessListElement::~ProcessListElement() in libdbms.a(Context.cpp.o)
      DB::UnionBlockInputStream<(DB::StreamUnionMode)0>::readImpl() in libdbms.a(StorageDistributed.cpp.o)
      DB::ParallelInputsProcessor<DB::UnionBlockInputStream<(DB::StreamUnionMode)0>::Handler, (DB::StreamUnionMode)0>::thread(MemoryTracker*, unsigned long) in libdbms.a(StorageDistributed.cpp.o)
      DB::UnionBlockInputStream<(DB::StreamUnionMode)1>::readImpl() in libdbms.a(InterpreterCheckQuery.cpp.o)
      ...
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[3]: *** [dbms/src/Server/clickhouse] Error 1
make[2]: *** [dbms/src/Server/CMakeFiles/clickhouse.dir/all] Error 2
make[1]: *** [dbms/src/Server/CMakeFiles/clickhouse.dir/rule] Error 2
make: *** [clickhouse] Error 2
```

The clang version:
```
$ clang -v
Apple LLVM version 9.0.0 (clang-900.0.38)
Target: x86_64-apple-darwin16.6.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
Found CUDA installation: /usr/local/cuda, version 8.0
```


Note that it goods well if I compile with gcc 7. i.e. 
```
cmake .. -DCMAKE_CXX_COMPILER=`which g++-7` -DCMAKE_C_COMPILER=`which gcc-7`
```

--- a/dbms/src/Common/MemoryTracker.cpp
+++ b/dbms/src/Common/MemoryTracker.cpp
@@ -149,8 +149,11 @@ void MemoryTracker::setOrRaiseLimit(Int64 value)
         ;
 }
 
-
+#if defined(__apple_build_version__) && __apple_build_version__ <= 9000038
+__thread MemoryTracker * current_memory_tracker = nullptr;
+#else
 thread_local MemoryTracker * current_memory_tracker = nullptr;
+#endif
 
 namespace CurrentMemoryTracker
 {

--- a/dbms/src/Common/MemoryTracker.h
+++ b/dbms/src/Common/MemoryTracker.h
@@ -108,7 +108,11 @@ public:
   * This pointer is set when memory consumption is monitored in current thread.
   * So, you just need to pass it to all the threads that handle one request.
   */
+#if defined(__apple_build_version__) && __apple_build_version__ <= 9000038
+extern __thread MemoryTracker * current_memory_tracker;
+#else
 extern thread_local MemoryTracker * current_memory_tracker;
+#endif
 
 /// Convenience methods, that use current_memory_tracker if it is available.
 namespace CurrentMemoryTracker

